lib/route.js

```diff
@@ -35,11 +35,13 @@ exports = module.exports = internals.Route = function (options, connection, real
     Hoek.assert(options.handler || (options.config && options.config.handler), 'Missing or undefined handler:', options.method, options.path);
     Hoek.assert(!!options.handler ^ !!(options.config && options.config.handler), 'Handler must only appear once:', options.method, options.path);            // XOR
     Hoek.assert(options.path === '/' || options.path[options.path.length - 1] !== '/' || !connection.settings.router.stripTrailingSlash, 'Path cannot end with a trailing slash when connection configured to strip:', options.method, options.path);
+    Hoek.assert(/^[a-zA-Z0-9!#\$%&'\*\+\-\.^_`\|~]+$/.test(options.method), 'Invalid method name:', options.method, options.path);
```

Comment:
Wouldn't it be faster and probably easier to read to just have a method object and reference that?

```js
var _methdods = { get: true, post: true...}
Hoek.assert(_methods[options.method.toLowerCase()])
```
or something along those lines?